File: ruby.vb.ns.com/scripts/example/branch.rb
#
# 条件分岐
#

=begin
true:  それ以外 (0 '' を含む)
false: false nil
=end

### if
puts "---- if ----"

i = 90

if i > 60
    puts "Hard"
elsif i > 40 && i < 60
    puts "Normal"
else
    puts "Easy"
end

# 省略系
puts "OK" if i > 60

# 三項演算子
puts i > 60 ? 'OK' : 'NG'

### case
puts "---- case ----"

color = "blue"

case color
when "blue", "green"
    puts "100"
when "red"
    puts "200"
when "yellow"
    puts "300"
else
    puts "0"
end

File: golang.vb.ns.com/scripts/example/branch.go
/**
 * 条件分岐
 */
package main

import "fmt"

func main() {
    i := 1

    // if
    fmt.Println("---- if ----")

    if i == 0 {
        fmt.Println("zero")
    } else if i == 1 {
        fmt.Println("one")
    } else {
        fmt.Println("not zero")
    }

    // ifの条件式の前に処理を記載することが可能です
    // ifの中だけで利用する局所変数を定義するのに便利です。
    if s := "blue"; s == "blue" {
        fmt.Println("GOOD")
    } else {
        fmt.Println("BAD")
    }

    // switch
    fmt.Println("---- switch ----")

    color := "blue"
    switch color {
        case "red":
            fmt.Println(300)
        case "blue", "green":
            fmt.Println(200)
        case "yellow":
            fmt.Println(100)
        default:
            fmt.Println(0)
    }
}
